app/controllers/concerns/publishable.rb

```diff
@@ -0,0 +1,35 @@
+module Publishable
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :ensure_reviewer, only: [:create, :update, :multiple_update]
+  end
+
+  private
+
+  def ensure_reviewer
+    return if can?(:publish, current_project)
+
+    record_params = self.send("#{controller_name.singularize}_params")
+    current_state = record&.state
+
+    if (current_state != 'published' && record_params[:state] == 'published') ||
```

Comment:
What case would we be using params instead of strong params?

Reply:
strong params for `issues#update` while params for `qa/issues#multiple_update`